android/app/src/test/java/org/smartregister/fhircore/api/HapiFhirServiceTest.kt

```diff
@@ -16,38 +16,41 @@
 
 package org.smartregister.fhircore.api
 
-import android.content.Context
 import androidx.test.core.app.ApplicationProvider
 import ca.uhn.fhir.context.FhirContext
-import io.mockk.every
-import io.mockk.mockk
-import io.mockk.mockkObject
+import java.lang.reflect.Proxy
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
 import org.junit.Assert
-import org.junit.Before
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.robolectric.annotation.Config
-import org.smartregister.fhircore.robolectric.FhircoreTestRunner
-import org.smartregister.fhircore.shadow.FhirApplicationShadow
+import org.robolectric.util.ReflectionHelpers
+import org.smartregister.fhircore.BuildConfig
+import org.smartregister.fhircore.RobolectricTest
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
 
-@RunWith(FhircoreTestRunner::class)
-@Config(shadows = [FhirApplicationShadow::class])
-class HapiFhirServiceTest {
-  private var mockService: HapiFhirService? = null
-  private val context = ApplicationProvider.getApplicationContext<Context>()
-  private val parser = FhirContext.forR4().newJsonParser()
+class HapiFhirServiceTest : RobolectricTest() {
 
-  @Before
-  fun setUp() {
-    mockService = mockk()
+  @Test
+  fun `create() should HapiFhirService with correct base uRL, client and converter factories`() {
```

Comment:
I think this test case can be renamed for better clarity

Reply:
Do you mean this `create() should return HapiFhirService with correct base url, client, interceptors and converter factories`

Reply:
Yeah